src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/LibraryImportGenerator.cs

```diff
@@ -70,7 +71,20 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                         ? new { Syntax = (MethodDeclarationSyntax)context.TargetNode, Symbol = methodSymbol }
                         : null)
                 .Where(
-                    static modelData => modelData is not null);
+                    modelData =>
```

Comment:
Do the Roslyn APIs guarantee the callback is invoked synchronously as part of invoking Where?  I expected this was all about setting up the pipeline and could be invoked subsequently.

Reply:
This method does just set up the pipeline. This implementation needs to determine when to emit the diagnostic as part of the pipeline steps.